vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/ElementSteps.java

```diff
@@ -315,6 +317,25 @@ public void clickOnAllElements(SearchAttributes locator)
                 .forEach(e -> mouseActions.click(e));
     }
 
+    /**
+     * Verifies elements' located by locator state.
+     * Where state one of: ENABLED/DISABLED, SELECTED/NOT_SELECTED, VISIBLE/NOT_VISIBLE
+     * @param state Desired state of an element
+     * @param locator Locator to locate element
+     * @param comparisonRule The rule to compare values
+     * (<i>Possible values:<b> LESS_THAN, LESS_THAN_OR_EQUAL_TO, GREATER_THAN, GREATER_THAN_OR_EQUAL_TO,
+     * EQUAL_TO</b></i>)
+     * @param quantity Desired amount of elements
+     */
+    @Then("number of $state elements found by `$locator` is $comparisonRule `$quantity`")
+    public void doesElementsNumberInStateExists(State state, SearchAttributes locator, ComparisonRule comparisonRule,
+            int quantity)
+    {
+        String description = "Element is " + state;
+        thePageContainsQuantityElements(locator, comparisonRule, quantity)
+                .forEach(e -> baseValidations.assertElementState(description, state, e));
```

Comment:
Use case:
```
Then number of visible elements found by `...` is equal to `3`
```

There are 4 elements by locator, and only 3 of them are visible, what will the step result be? I assume failure

Reply:
Yep as stated - equal to 3 visible elements expected but got 4: 3 visible, and 1 invisible.
1. Fail condition check
2. Fail visibility check